### PR TITLE
Retrieve value with trie depth

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- In order to have an adaptive gas model based on how many levels a trie has, getting values from data tries should also return the depth from which the value is retrieved
  
## Proposed Changes
- When getting values from dataTries, also return the value from which that value was retrieved.

## Testing procedure
- Normal testing procedure

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,16 +4,16 @@ go 1.15
 
 require (
 	github.com/ElrondNetwork/covalent-indexer-go v1.0.6
-	github.com/ElrondNetwork/elastic-indexer-go v1.2.44
-	github.com/ElrondNetwork/elrond-go-core v1.1.20
+	github.com/ElrondNetwork/elastic-indexer-go v1.2.45-0.20221019144818-5b4fe7b3e724
```
Comment (ssd04) [inline go.mod:7]: note to not forget about proper release tags

File: state/trackableDataTrie.go [hunk 2]
```diff
@@ -26,27 +26,28 @@ func NewTrackableDataTrie(identifier []byte, tr common.Trie) *trackableDataTrie
...
-		return nil, ErrNilTrie
+		return nil, 0, ErrNilTrie
 	}
-	value, err := tdaw.tr.Get(key)
+	value, maxDepth, err := tdaw.tr.Get(key)
 	if err != nil {
-		return nil, err
+		return nil, maxDepth, err
 	}
 	log.Trace("retrieve value from trie", "key", key, "value", value)
```
Comment (ssd04) [inline state/trackableDataTrie.go:47]: should we think to log alse the maxDepth value?

File: trie/patriciaMerkleTrie.go [hunk 3]
```diff
@@ -80,22 +80,22 @@ func NewTrie(
...
 	defer tr.mutOperation.Unlock()
 
 	if tr.root == nil {
-		return nil, nil
+		return nil, 0, nil
 	}
 	hexKey := keyBytesToHex(key)
 
-	val, err := tr.root.tryGet(hexKey, tr.trieStorage)
+	val, maxDepth, err := tr.root.tryGet(hexKey, rootDepthLevel, tr.trieStorage)
```
Comment (AdoAdoAdo) [inline trie/patriciaMerkleTrie.go:92]: also depth here?

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: state/trackableDataTrie.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply state/trackableDataTrie.go:47]: Done.

File: state/trackableDataTrie.go [hunk 4]
```diff
@@ -26,27 +26,28 @@ func NewTrackableDataTrie(identifier []byte, tr common.Trie) *trackableDataTrie
...
+		return trimmedVal, 0, err
 	}
 
 	// ok, not in cache, retrieve from trie
 	if tdaw.tr == nil {
-		return nil, ErrNilTrie
+		return nil, 0, ErrNilTrie
 	}
-	value, err := tdaw.tr.Get(key)
+	value, maxDepth, err := tdaw.tr.Get(key)
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/trackableDataTrie.go:43]: Done.

File: trie/patriciaMerkleTrie.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply trie/patriciaMerkleTrie.go]: Done.

**Review by gabi-vuls — DISMISSED** [review]
system test passed.

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**